src/Approximations/overapproximate.jl

```diff
@@ -516,3 +516,93 @@ function overapproximate(cap::Intersection{N,
     return overapproximate(swap(cap), dir; kwargs...)
 end
 
+"""
+    overapproximate(lm::LinearMap{N, <:CartesianProductArray{N, <:LazySet{N}}},
+                         dir::Type{<:AbstractDirections})::CartesianProductArray{N, HPolytope{N}} where {N}
+
+Overapproximating a lazy linear map of cartesian product array with template directions for each block.
+
+### Input
+
+- `lm`   -- lazy linear map of cartesian product array
+- Type{<:CartesianProductArray} -- type to specify decomposed overapproximation
+- `oa` -- (concrete) direction representation
+
+### Output
+
+An `CartesianProductArray` with overapproximation of the each block with the directions from `dir`.
+"""
+function overapproximate(lm::LinearMap{N, <:CartesianProductArray{N,
+                            <:LazySet{N}}}, ::Type{<:CartesianProductArray}, oa::Union{Type{<:AbstractDirections},Type{<:Hyperrectangle}, Type{<:Interval}}=Hyperrectangle) where {N}
+
+    if !(size(lm.M, 2) == dim(lm.X))
+        error("Matrix needs to be commensurate with the cartesian product")
+    end
+
+    cp = lm.X
+    M = lm.M
+
+    col_st_index, col_end_ind = 0, 0
+
+    return decomposed_overapproximation(cp, M, col_st_index, col_end_ind, oa)
+end
+
+#template directions
+function decomposed_overapproximation(cp::CartesianProductArray{N,<:LazySet{N}}, M::AbstractArray{N},
+            col_st_index::Int, col_end_ind::Int, oa::Type{<:AbstractDirections}) where {N}
+
+    array =  Vector{HPolytope{N}}()
+
+    sizehint!(array,length(cp.array))
+
+    for bi in cp.array
+        v_block, col_st_index, col_end_ind = overapproximate_block(bi, col_st_index, col_end_ind, M)
+
+        push!(array, Approximations.overapproximate(v_block, oa{N}(dim(v_block))))
+    end
+
+    result = CartesianProductArray{N, HPolytope{N}}(array)
+    return result
+end
+
+#Interval Hull
+function decomposed_overapproximation(cp::CartesianProductArray{N,<:LazySet{N}}, M::AbstractArray{N},
+            col_st_index::Int, col_end_ind::Int, oa::Union{Type{<:Hyperrectangle}, Type{<:Interval}}) where {N}
+
+    array =  Vector{oa{N}}()
+
+    sizehint!(array,length(cp.array))
+
+    for bi in cp.array
+        v_block, col_st_index, col_end_ind = overapproximate_block(bi, col_st_index, col_end_ind, M)
+
+        push!(array, Approximations.overapproximate(v_block, oa))
+    end
+
+    result = CartesianProductArray{N, oa{N}}(array)
+    return result
+end
+
+#overapproximation of linear map to each block
+function overapproximate_lm_block(bi:: LazySet{N}, col_st_index::Int, col_end_ind::Int, M::AbstractArray{N}) where {N}
+    col_st_index = col_end_ind + 1
+    n = dim(bi)
+    col_end_ind += n
+
+    start_ind, end_ind = 1, n
```

Comment:
```suggestion
    row_start_ind, row_end_ind = 1, n
```
(helps in understanding)